helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -376,62 +614,158 @@ public Object removeRoutingTableChangeListener(
    * @return
    */
   public Collection<LiveInstance> getLiveInstances() {
-    return _routingTableRef.get().getLiveInstances();
+    String key = getRoutingTableKey();
+    return _routingTableRef.get(key).get().getLiveInstances();
+  }
+
+  public Collection<LiveInstance> getLiveInstances(PropertyType propertyType) {
+    String key = getRoutingTableKey(propertyType);
+    return _routingTableRef.get(key).get().getLiveInstances();
+  }
+
+  public Collection<LiveInstance> getLiveInstances(PropertyType propertyType, String type) {
+    String key = getRoutingTableKey(propertyType, type);
+    return _routingTableRef.get(key).get().getLiveInstances();
   }
 
   /**
    * Return all instance's config in this cluster.
    * @return
    */
   public Collection<InstanceConfig> getInstanceConfigs() {
-    return _routingTableRef.get().getInstanceConfigs();
+    String key = getRoutingTableKey();
+    return _routingTableRef.get(key).get().getInstanceConfigs();
+  }
+
+  public Collection<InstanceConfig> getInstanceConfigs(PropertyType propertyType) {
```

Comment:
Same here, instance config would be different with a different type?

Reply:
Removed this method.

Reply:
Ok I picked one of the keys and merge all of them to one API.